gene/delete.go

```diff
@@ -0,0 +1,338 @@
+package gene
+
+import (
+	"errors"
+	"github.com/vertgenlab/gonomics/dna"
+)
+
+var (
+	ErrNoStopFound = errors.New("frameshift was unable to find a stop before the end of the gene")
+)
+
+//TODO EffectPrediction
+// Deletion removes bases from the Gene, predicts the effect, and updates the Gene struct to reflect the change.
+// genomeStartPos should be the base directly BEFORE the deleted bases. genomeEndPos should be the base directly AFTER the deleted bases.
+// All positions should be given as base-zero genomic coordinates.
+func Deletion(g *Gene, genomeStartPos int, genomeEndPos int) (EffectPrediction, error) {
+	var answer EffectPrediction
+	var err error
+
+	genomeIndexStartPos, genomeIndexEndPos, err := deletionPreRunChecks(g, genomeStartPos, genomeEndPos)
+	if err != nil {
+		return EffectPrediction{}, err
+	}
+
+	// Save CDS start and stop for later
+	origCdsStart := g.cdsStarts[0]
+	origCdsEnd := g.cdsEnds[len(g.cdsEnds)-1]
+
+	// Delete from Genome Sequence
+	copy(g.genomeSeq[genomeIndexStartPos+1:], g.genomeSeq[genomeIndexEndPos:])
+	g.genomeSeq = g.genomeSeq[:len(g.genomeSeq)-(genomeIndexEndPos-(genomeIndexStartPos+1))]
+
+	// Update startPos if necessary
+	if genomeIndexStartPos == -1 {
+		g.startPos += genomeIndexEndPos - (genomeIndexStartPos + 1)
```

Comment:
I am struggling with this if statement.  I am thinking that genomeIndexStartPos is the genomic start of the deletion mapped into a coordinate system where the TSS is 0 and the second transcripted base (in whatever direction) is 1.  So now we have an index of -1, which would be the base just before the TSS.  Because we are left-open, we are starting the deletion by deleting the TSS, and we do:
g.startPos += genomeIndexEndPos - (-1 + 1)
evaluate to:
g.startPos += genomeIndexEndPos
Am I thinking about this right?  Is it more clear without the extra math that evals to zero?

Reply:
You are correct, that simplifies things since in this case we already know that genomeIndexStartPos is zero (since updating to left closed).